File: core/src/remove/tanks/game/physics/fixture/FixturePrefab.java
package remove.tanks.game.physics.fixture;

import com.google.common.base.MoreObjects;
import remove.tanks.game.physics.filter.FilterPrefab;
import remove.tanks.game.physics.shape.ShapePrefab;
import remove.tanks.game.utility.Prefab;

/**
 * @author Mateusz Długosz
 */
public final class FixturePrefab extends Prefab {
    private final FilterPrefab filterPrefab;
    private final ShapePrefab shapePrefab;
    private final float restitution;
    private final float density;
    private final float friction;

    public FixturePrefab(
            FilterPrefab filterPrefab,
            ShapePrefab shapePrefab,
            float restitution,
            float density,
            float friction
    ) {
        this.filterPrefab = filterPrefab;
        this.shapePrefab = shapePrefab;
        this.restitution = restitution;
        this.density = density;
        this.friction = friction;
    }

    public FilterPrefab getFilterPrefab() {
        return filterPrefab;
    }

    public ShapePrefab getShapePrefab() {
        return shapePrefab;
    }

    public float getRestitution() {
        return restitution;
    }

    public float getDensity() {
        return density;
    }

    public float getFriction() {
        return friction;
    }

    @Override
    public String toString() {
        return MoreObjects.toStringHelper(this)
                .add("filterPrefab", filterPrefab)
                .add("shapePrefab", shapePrefab)
                .add("restitution", restitution)
                .add("density", density)
                .add("friction", friction)
                .toString();
    }
}

File: tools/docs-generator/lib/physics/fixture/fixture_prefab.py
class FixturePrefab(object):
    def __init__(self, filter_prefab, shape_prefab, restitution, density, friction, is_sensor):
        self.filter_prefab = filter_prefab
        self.shape_prefab = shape_prefab
        self.restitution = float(restitution)
        self.density = float(density)
        self.friction = float(friction)
        self.is_sensor = bool(is_sensor)

    def get_filter_prefab(self):
        return self.filter_prefab

    def get_shape_prefab(self):
        return self.shape_prefab

    def get_restitution(self):
        return self.restitution

    def get_density(self):
        return self.density

    def get_friction(self):
        return self.friction

    def get_is_sensor(self):
        return self.is_sensor

    def __str__(self):
        return "FixturePrefab(filter_prefab={}, shape_prefab={}, restitution={}, density={}, friction={}, is_sensor={})"\
            .format(self.filter_prefab, self.shape_prefab, self.restitution, self.density, self.friction, self.is_sensor)
